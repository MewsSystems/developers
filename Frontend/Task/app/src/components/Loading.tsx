import React from "react";
import styles from "../style.module.css";

const Loading: React.FC<{ message: string }> = ({ message }) => {
  return (
    <div className={styles.center}>
      <div>{message}</div>
      <div className={styles.ldsCircle}>
        <div></div>
      </div>
    </div>
  );
};

export default Loading;
