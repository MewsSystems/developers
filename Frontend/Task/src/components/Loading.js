import React, { Component } from 'react';

class Loading extends Component {
  render() {
    const { className } = this.props;
    return (
      <div className={`loading ${className || ''}`}>
        <i className="fas fa-spinner" />
      </div>
    );
  };
};

export default Loading;