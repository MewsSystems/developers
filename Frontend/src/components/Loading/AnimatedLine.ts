import styled from 'styled-components'

export interface AnimatedLineProps {
  width: string
  height: string
}

export const AnimatedLine = styled.div<AnimatedLineProps>`
  width: ${({ width }) => width};
  height: ${({ height }) => height};
  background: linear-gradient(
    90deg,
    rgba(195, 195, 195, 1) 0%,
    rgba(246, 246, 246, 1) 100%
  );
  animation: AnimationName 5s ease infinite;

  @keyframes AnimationName {
    0% {
      background-position: 0% 50%;
    }
    50% {
      background-position: 100% 50%;
    }
    100% {
      background-position: 0% 50%;
    }
  }
`
