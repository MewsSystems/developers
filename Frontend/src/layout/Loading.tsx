import * as React from 'react'
import { keyframes } from '@emotion/core'
import { transparentize } from 'polished'
import styled from '../utils/styled'

export const LoadingSpinner: React.SFC = () => (
  <Spinner>
    <div className="cube1" />
    <div className="cube2" />
  </Spinner>
)

const CubeMove = keyframes`
  25% {
    transform: translateX(42px) rotate(-90deg) scale(0.5);
    -webkit-transform: translateX(42px) rotate(-90deg) scale(0.5);
  } 50% {
    transform: translateX(42px) translateY(42px) rotate(-179deg);
    -webkit-transform: translateX(42px) translateY(42px) rotate(-179deg);
  } 50.1% {
    transform: translateX(42px) translateY(42px) rotate(-180deg);
    -webkit-transform: translateX(42px) translateY(42px) rotate(-180deg);
  } 75% {
    transform: translateX(0px) translateY(42px) rotate(-270deg) scale(0.5);
    -webkit-transform: translateX(0px) translateY(42px) rotate(-270deg) scale(0.5);
  } 100% {
    transform: rotate(-360deg);
    -webkit-transform: rotate(-360deg);
  }
`

const Spinner = styled('div')`
  margin: 100px auto;
  width: 40px;
  height: 40px;
  position: relative;

  & .cube1,
  & .cube2 {
    background-color: ${props => props.theme.colors.brand};
    width: 15px;
    height: 15px;
    position: absolute;
    top: 0;
    left: 0;

    -webkit-animation: ${CubeMove} 1.8s infinite ease-in-out;
    animation: ${CubeMove} 1.8s infinite ease-in-out;
  }

  & .cube2 {
    -webkit-animation-delay: -0.9s;
    animation-delay: -0.9s;
  }
`

export const LoadingOverlay = styled('div')`
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 10;
  background: ${props => transparentize(0.25, props.theme.colors.background)};
`

export const LoadingOverlayInner = styled('div')`
  display: flex;
  align-items: center;
  justify-content: center;
  min-height: 200px;
`

type LoadingProps = {
  loading: boolean
}

export const Loading = ({ loading }: LoadingProps) => {
  if (!loading) return null

  return (
    <LoadingOverlay>
      <LoadingOverlayInner>
        <LoadingSpinner />
      </LoadingOverlayInner>
    </LoadingOverlay>
  )
}
