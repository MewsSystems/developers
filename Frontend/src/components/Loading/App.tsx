import React from 'react'
import styled from 'styled-components'
import { COLORS } from 'constants/colors'
import { Content } from 'components/Content'
import { AnimatedLine } from './AnimatedLine'
import { Spacing } from 'components/Spacing'

const Header = styled.div`
  height: 10.6rem;
  background: ${COLORS.GRAY};
`

export const App: React.FC = () => (
  <>
    <Header />
    <Content>
      <Spacing outer="2rem 0">
        <AnimatedLine width="20rem" height="2rem" />
      </Spacing>
    </Content>
  </>
)
