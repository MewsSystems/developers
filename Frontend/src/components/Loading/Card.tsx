import React from 'react'
import { AnimatedLine } from './AnimatedLine'
import { StyledCard } from 'components/Card'
import { Spacing } from 'components/Spacing'

export const Card: React.FC = () => (
  <StyledCard>
    <Spacing outer="0 0 1rem">
      <AnimatedLine width="100%" height="2rem" />
    </Spacing>
    <AnimatedLine width="100%" height="1rem" />
  </StyledCard>
)
