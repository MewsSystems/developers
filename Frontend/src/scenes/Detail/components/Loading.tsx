import React from 'react'
import { StyledHeroBanner } from 'components/HeroBanner'
import { Content } from 'components/Content'
import { AnimatedLine } from 'components/Loading/AnimatedLine'

export const Loading: React.FC = () => (
  <>
    <StyledHeroBanner />
    <Content>
      <AnimatedLine height="2rem" width="80%" />
    </Content>
  </>
)
