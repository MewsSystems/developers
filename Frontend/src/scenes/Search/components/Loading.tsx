import React from 'react'
import { Content } from 'components/Content'
import { Spacing } from 'components/Spacing'
import styled from 'styled-components'
import { Card } from 'components/Loading/Card'
import { AnimatedLine } from 'components/Loading/AnimatedLine'

const Grid = styled.div`
  display: grid;
  grid-gap: 1rem;
  grid-template-columns: 1fr 1fr 1fr;
`

export const Loading: React.FC = () => (
  <Content>
    <Spacing outer="2rem 0">
      <AnimatedLine width="16rem" height="2rem" />
    </Spacing>
    <Spacing outer="2rem 0">
      <AnimatedLine width="12rem" height="1rem" />
    </Spacing>
    <Grid>
      <Card />
      <Card />
      <Card />
    </Grid>
  </Content>
)
