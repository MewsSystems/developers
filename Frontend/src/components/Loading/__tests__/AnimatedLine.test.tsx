import { AnimatedLine } from 'components/Loading/AnimatedLine'
import { render } from 'enzyme'
import 'jest-styled-components'
import React from 'react'

describe('Test Loading AnimatedLine', () => {
  it('Tests snapshot', () => {
    expect(
      render(<AnimatedLine width="10rem" height="10rem" />)
    ).toMatchSnapshot()
  })
})
