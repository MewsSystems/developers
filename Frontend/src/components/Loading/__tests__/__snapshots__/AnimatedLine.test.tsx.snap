// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Loading AnimatedLine Tests snapshot 1`] = `
.c0 {
  width: 10rem;
  height: 10rem;
  background: linear-gradient( 90deg, rgba(195,195,195,1) 0%, rgba(246,246,246,1) 100% );
  -webkit-animation: AnimationName 5s ease infinite;
  animation: AnimationName 5s ease infinite;
}

<div
  class="c0"
  height="10rem"
  width="10rem"
/>
`;
